### PR TITLE
drivers: counter: esp32: add support for esp32c3 to counter unified driver

### PR DESCRIPTION
**Pull request by uLipe**
This PR adds the ESP32C3 as supported SoC to the esp32 timer/counter unified driver.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: boards/riscv/esp32c3_devkitm/esp32c3_devkitm_defconfig [hunk 1]
```diff
@@ -14,3 +14,5 @@ CONFIG_PINMUX_ESP32=y
 CONFIG_GPIO=y
 CONFIG_GPIO_ESP32=y
 CONFIG_CLOCK_CONTROL=y
+CONFIG_COUNTER=y
+CONFIG_COUNTER_ESP32=y
```
Comment (sylvioalves) [inline boards/riscv/esp32c3_devkitm/esp32c3_devkitm_defconfig:18]: I think you can remove this from every esp32 soc as this shall be set automatically from now on.

Comment (uLipe) [reply boards/riscv/esp32c3_devkitm/esp32c3_devkitm_defconfig]: I removed then, thanks @sylvioalves

**Review by sylvioalves — APPROVED** [review]

**Comment by uLipe** [comment]
@cfriedt , @nashif the ci errors seem to not be related with the contents of this PR.

**Comment by uLipe** [comment]
Just rebased

**Comment by uLipe** [comment]
Folks, it seems the CI errors were gone after rebasing, shall we move forward with this PR?

**Comment by uLipe** [comment]
Folks, can we move forward with this PR? CI errors have been gone now :)

**Pull request merged**